tools/topology2/include/common/connection.conf

```diff
@@ -0,0 +1,32 @@
+Class.Base."connection" {
+
+	@args."type" {
+		type "string"
+	}
+
+	@args."pipeline_id" {
+		type "integer"
+	}
+
+	@args."index" {
+		type "integer"
+	}
+
+	DefineAttribute."source" {
+		constraints {
```

Comment:
I think the mask is easier to understand better when its within the attribute definition instead of clubbing all attributes into the  "mask" attribute. Let me experiment a bit more and see if I can make this easier

Reply:
I've changed to add attributes{} at the top-level.